orchestrator/graphql/resolvers/process.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2019-2020 SURF.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module that implements process related API endpoints."""
+
+import struct
+import zlib
+from enum import Enum
+from http import HTTPStatus
+from typing import List, Union
+from uuid import UUID
+
+import strawberry
+import structlog
+from more_itertools import chunked, flatten, one
+from sqlalchemy import String, cast
+from sqlalchemy.orm import defer, joinedload
+from sqlalchemy.sql import expression
+from strawberry.types.nodes import SelectedField, Selection
+
+from orchestrator.api.error_handling import raise_status
+from orchestrator.api.helpers import VALID_SORT_KEYS
+from orchestrator.db import ProcessSubscriptionTable, ProcessTable, ProductTable, SubscriptionTable, db
+from orchestrator.graphql.schemas.process import Process
+from orchestrator.graphql.types import CustomInfo
+from orchestrator.schemas import ProcessSchema
+from orchestrator.schemas.process import ProcessStepSchema
+
+Cursor = int
+logger = structlog.get_logger(__name__)
+
+
+def get_selections(selected_field: Selection) -> list[Selection]:
```

Comment:
will add docstring for the usage, Its to easily get the selected fields of the request, which we can use to only fetch those fields from the database instead of everything.